src/composables/nav.ts

```diff
@@ -21,3 +29,204 @@ export interface ActionItem {
   label: string
   show?: boolean | ((...args: any[]) => boolean)
 }
+
+export interface UseTabHistoryOpts {
+  /**
+   * The initial value of activeTab.
+   * The default value is tabs[0].value.
+   *
+   * When a valid tab value is found on window.location.search
+   * it will replace the initial value.
+   */
+  initial?: string
+
+  /**
+   * The param name to use on window.location.search.
+   * The default is "tab".
+   */
+  paramName?: string
+}
+
+/**
+ * useTabHistory
+ *
+ * Example Usage:
+ *
+ * const {activeTab, tabs} = useTabHistory([{label: "Tab One", value: "tab-1"}, {label: "Tab Two", value: "tab-2"}]})
+ *
+ * <Tabs v-model="activeTab" :tabs="tabs" />
+ *
+ * <div v-if="activeTab === 'tab-1'">Tab 1 Content</div>
+ * <div v-if="activeTab === 'tab-2'">Tab 2 Content</div>
+ *
+ * @param opts UseTabHistoryOpts
+ * @return {activeTab: Ref<string>, tabs: Ref<{labe: string, value: string}[]>}
+ */
+export const useTabHistory = (
+  initialTabs: MaybeRef<
+    {
+      label: string
+      value: string
+    }[]
+  >,
+  opts?: UseTabHistoryOpts
+) => {
+  const tabs = ref(initialTabs)
+  const config = {
+    initial: tabs?.value[0]?.value || "",
+    paramName: "tab",
+    ...opts,
+  }
+  const activeTab = ref(config.initial)
+
+  const isTab = (tab: string) => {
+    return tabs.value.find((t) => t.value === tab) ? true : false
+  }
+
+  // Kick off history tracking by applying the existing value
+  // on window.location.search if it is a valid tab option.
+  const params = new URLSearchParams(window.location.search)
+  const initialParam = params.get(config.paramName)
+
+  if (initialParam && isTab(initialParam)) {
+    activeTab.value = initialParam
+  }
+
+  watch(activeTab, (tab) => {
+    const params = new URLSearchParams(window.location.search)
+    params.set(config.paramName, tab.toString())
+
+    const queryString = params.toString() !== "" ? `?${params.toString()}` : ""
+
+    window.history.replaceState(
+      { tab: tab },
+      document.title,
+      `${window.location.pathname}${queryString}`
+    )
+  })
+
+  return {
+    activeTab,
```

Comment:
I expected to see an exported func to check if the tab is active. I am not doing anything more in archive than checking `isActiveTab`. Worth exporting the function to avoid `activeTab == 'tab-1'` as a 3rd export?

Reply:
Yep, not opposed to this - even had a moment where it was included, but dropped it.  I didn't see it at first, but I suppose `v-if="isActiveTab('tab-1')"` in a template is a bit easier on the eye's than `v-if="activeTab === 'tab-1'"`.

Is that what you're looking for here?

I couldn't quite get it nailed down, but there's a future where an `isActiveTab('my-tab')` method can also be type safe on `tabs[0].value` values, but all implementations led to the use of `as const` which created more headaches, and it ultimately didn't seem helpful enough to pursue.

Reply:
Yeah exactly `isActiveTab('my-tab')`. It's a small win, but I find myself writing it in each component for the ergonomics

Reply:
@profsmallpine should we go one more step and updated this composable to `useTabNav(tabs, config)`, where `useHistory` is a config option?  That would expose `isActiveTab` in situations where having a tab history isn't desirable.

Reply:
At this point, I would want this to push to the browser's history in all use cases and I can't picture a use case where we would want to exclude that as a rule. I'm sure it exists, but I would probably start without so it's intentional. Not sure on the balance of predicting the future so that minor changes to usage forces a trees bump, but as you know, I default to less functionality at the start.